src/Controls/src/Core/FontElement.cs

```diff
@@ -108,14 +116,36 @@ static void OnFontAttributesChanged(BindableObject bindable, object oldValue, ob
 			var fontSize = (double)bindable.GetValue(FontSizeProperty);
 			var fontAttributes = (FontAttributes)newValue;
 			var fontFamily = (string)bindable.GetValue(FontFamilyProperty);
+			var enableAutoScaling = (bool)bindable.GetValue(FontScalingEnableProperty);
 
 			if (fontFamily != null)
-				bindable.SetValue(FontProperty, Font.OfSize(fontFamily, fontSize).WithAttributes(fontAttributes));
+				bindable.SetValue(FontProperty, Font.OfSize(fontFamily, fontSize, enableScaling: enableAutoScaling).WithAttributes(fontAttributes));
 			else
-				bindable.SetValue(FontProperty, Font.SystemFontOfSize(fontSize).WithAttributes(fontAttributes));
+				bindable.SetValue(FontProperty, Font.SystemFontOfSize(fontSize, enableScaling: enableAutoScaling).WithAttributes(fontAttributes));
 
 			SetCancelEvents(bindable, false);
 			((IFontElement)bindable).OnFontAttributesChanged((FontAttributes)oldValue, (FontAttributes)newValue);
 		}
+
+		static void OnEnableFontScalingChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			if (GetCancelEvents(bindable))
+				return;
+
+			SetCancelEvents(bindable, true);
+
+			var fontSize = (double)bindable.GetValue(FontSizeProperty);
+			var fontAttributes = (FontAttributes)bindable.GetValue(FontAttributesProperty);
+			var fontFamily = (string)bindable.GetValue(FontFamilyProperty);
+			var enableAutoScaling = (bool)newValue;
+
+			if (fontFamily != null)
+				bindable.SetValue(FontProperty, Font.OfSize(fontFamily, fontSize, enableScaling: enableAutoScaling).WithAttributes(fontAttributes));
+			else
+				bindable.SetValue(FontProperty, Font.SystemFontOfSize(fontSize, enableScaling: enableAutoScaling).WithAttributes(fontAttributes));
+
+			SetCancelEvents(bindable, false);
```

Comment:
Is it not possible to refactor all these methods into a single one and call it? I worry we might make a change in one place and forget another, also will delete like 50 lines of code.

Reply:
Probably, I just went for the quick version right now and I'll look at refactoring this tomorrow

Reply:
Cleaned up a little bit. There's still a bit more cleanup that could happen here but I'll leave that for a future PR